mp2i/cogs/sanctions.py

```diff
@@ -21,11 +25,33 @@ class Sanction(Cog):
     def __init__(self, bot):
         self.bot = bot
 
+    def __register_sanction_in_database(
+        self,
+        staff: int,
+        member: int,
+        guild: int,
+        date,
+        type_: str,
+        duration: Optional[int],
+        reason: Optional[str],
+    ):
+        database.execute(
+            insert(SanctionModel).values(
+                by_id=staff,
+                to_id=member,
+                guild_id=guild,
+                date=date,
+                type=type_,
+                duration=duration,
+                reason=reason,
+            )
+        )
+
     @hybrid_command(name="warn")
     @guild_only()
     @has_any_role("Modérateur", "Administrateur")
-    async def warn(self, ctx, member: discord.Member, dm: str, visible: str, *,
-                   reason: Optional[str]) -> None:  # fmt: skip
+    async def warn(self, ctx, member: discord.Member, dm: str, *,
```

Comment:
```suggestion
    @choices(dm=[Choice(name="oui", value="oui"), Choice(name="non", value="non")])
    async def warn(self, ctx, member: discord.Member, dm: str, *,
```
Ajouter aussi les bons imports

Reply:
On peut même faire en sorte que dm soit un booléen, ce serait même mieux